packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts

```diff
@@ -134,6 +135,61 @@ function removeSegmentationRepresentation(
   }
 }
 
+/**
+ * Checks if a segmentation data belongs to the same study as the series
+ * displayed in a given viewport
+ * @param viewport
+ * @param referencedVolumeId volume id of the segmentation reference series
+ * @returns
+ */
+function canRenderSegmentationInViewport(
```

Comment:
I think toolGroups are the solution for the need to restrict visualization of the segmentations on any viewport. @wayfarer3130 What do you think? We are breaking the contract between toolGroup's viewports and Segmentation by introducing this flag. It is very odd that there will be a viewport in a toolGroup in the same FrameOfReference that don't render the SEG

Reply:
That isn't really possible to do because the tool groups aren't defined per study right now in a way that is actually controllable.  It is a nice theory, I just don't see how to actually manage it.  If the tool group had a setting as part of the segmentation visualization that said same study instance UID, then I would agree, it would work and is probably the right approach.

Also, the images should absolutely require the same frame of reference to be displayed, I'd like that to be tested/confirmed that the same FOR is being provided for different studies images, and that it was intentional, and not accidental.  Really, the same FOR means it is actually the same coordinates, so you can actually display it.

Reply:
Something to always consider is that CS3D don't have drag and drop. CS3D is a rendering and tooling library. Any drag and drop related logic is an app concern.

Reply:
It doesn't matter how one chooses to display a new series, or even if you simply load multiple segmentations initially - the segmentation display is just not valid if it isn't the same FOR, so the labels shouldn't display.